src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunctionImpl.cs

```diff
@@ -319,7 +324,80 @@ public void Visit(TimeValue timeValue)
 
                 public void Visit(UntypedObjectValue untypedObjectValue)
                 {
-                    throw new ArgumentException($"Unable to serialize type {untypedObjectValue.GetType().FullName} to Json format.");
+                    Visit(untypedObjectValue.Impl);
+                }
+
+                private void Visit(IUntypedObject untypedObject, int index = 0)
+                {
+                    FormulaType type = untypedObject.Type;
+
+                    if (index > _maxDepth)
+                    {
+                        IRContext irContext = IRContext.NotInSource(FormulaType.UntypedObject);
+                        ErrorValues.Add(new ErrorValue(irContext, new ExpressionError()
+                        {
+                            ResourceKey = TexlStrings.ErrReachedMaxJsonDepth,
+                            Span = irContext.SourceContext,
+                            Kind = ErrorKind.InvalidArgument                            
+                        }));
+                        _writer.WriteStringValue(string.Empty);
+                        return;
+                    }
+
+                    if (type is StringType)
+                    {
+                        _writer.WriteStringValue(untypedObject.GetString());
+                    }
+                    else if (type is DecimalType || (type is ExternalType et && et.Kind == ExternalTypeKind.UntypedNumber))
```

Comment:
UntypedNumber needs to be serialized unmodified.  These tests are currently failing:

```
>> JSON(ParseJSON("1e300"))
One of the identified items was in an invalid format.

>> JSON(ParseJSON("1111111111111111111111111111111.2222222222222222222222222222222222"))
One of the identified items was in an invalid format.
```
Even with the super long decimal, it should be preserved.  We don't know if it is float or decimal until it is used.